tools/sof-dump-status.py

```diff
@@ -366,8 +402,15 @@ def dump_dapm(dapm, filter = "all"):
     sysinfo = clsSYSCardInfo()
     if ret_args['platform'] is True:
         sysinfo.loadPCI()
+        mach_name = None
         for pci_info in sysinfo.pci_lst:
-            print(pci_info['hw_name'])
+            mach_name = pci_info['hw_name']
+        if (mach_name is None):
+            sysinfo.loadACPI()
+            for acpi_info in sysinfo.acpi_lst:
+                mach_name = acpi_info['hw_name']
```

Comment:
Same comment: why the for loop instead of accessing the last element directly?

Reply:
don't ask  me, it's the same pattern everywhere.

Reply:
Copy/paste of code that doesn't make sense goes nowhere fast.